frontend/src/views/Collection/index.tsx

```diff
@@ -144,14 +149,31 @@ const Collection: FC<Props> = ({ id = "" }) => {
               Download
             </Button>
           </Popover>
-          <Button icon={IconNames.TRASH} minimal>
-            {/* TEMP: To show selected file */}
-            {/* TEMP: To show selected file */}
-            {/* TEMP: To show selected file */}
-            {selected}
-          </Button>
+          <Button
+            icon={IconNames.TRASH}
+            minimal
+            onClick={() => {
+              if (selected) openAlert(!isAlertOpen);
+            }}
+          ></Button>
         </StyledDiv>
       )}
+      <StyledAlert
+        cancelButtonText={"Cancel"}
+        confirmButtonText={"Delete Dataset"}
+        intent={Intent.DANGER}
+        isOpen={isAlertOpen}
+        onCancel={() => {
+          openAlert(!isAlertOpen);
+        }}
+        onConfirm={() => {
+          deleteDataset(selected ?? "");
```

Comment:
Also perhaps `deleteDataset()` could loosen its argument to accept an optional string, so we don't need to pass `""` at callsite?

Reply:
Technically, `deleteDataset()` should never be called with no collection attribution as the delete action can only be enacted when there is a collection.  The nullish coalescing is just to meet the needs at compile time

Reply:
Ah yeah, I'm thinking it's probably better to handle `""` or `undefined` in `deleteDataset()`, otherwise by passing `""` to the function and not handle the edge case, we would end up sending a malformed request to BE?

e.g., 

```ts
async function deleteDataset(
  dataset_uuid?: string // update type here to allow undefined to come through
): Promise<DatasetUploadStatus> {
  // handle edge case here
  if (!dataset_uuid) return Promise.reject('`dataset_uuid` is not valid')

  // real API call
}
```